app/src/components/panels/SupportProposal.js

```diff
@@ -0,0 +1,197 @@
+              false,
+              false,
+              {
+                commas: !editMode,
+                replaceZeroBy: editMode ? '' : '0',
+                rounding: stakeToken.tokenDecimals,
+              }
+            )
+          : '',
+      }))
+    },
+    [stakeToken]
+  )
+
+  // Amount change handler
+  const handleAmountChange = useCallback(
+    event => {
+      const newAmount = event.target.value
+
+      const newAmountBN = new BN(
+        isNaN(event.target.value)
+          ? -1
+          : toDecimals(newAmount, stakeToken.tokenDecimals)
+      )
+
+      setAmount({
+        value: newAmount,
+        valueBN: newAmountBN,
+      })
+    },
+    [stakeToken]
+  )
+
+  const handleMaxSelected = useCallback(() => {
+    setAmount({
+      valueBN: nonStakedTokens,
+      value: formatTokenAmount(
+        nonStakedTokens,
+        stakeToken.tokenDecimals,
+        false,
+        false,
+        { commas: false, rounding: stakeToken.tokenDecimals }
+      ),
+    })
+  }, [nonStakedTokens, stakeToken])
+
+  // Form submit handler
+  const handleSubmit = useCallback(
+    event => {
+      event.preventDefault()
+
+      api.stakeToProposal(id, String(amount.valueBN)).toPromise()
+
+      onDone()
+    },
+    [amount, api, onDone, stakeToken]
+  )
+
+  const errorMessage = useMemo(() => {
+    if (amount.valueBN.eq(new BN(-1))) {
+      return 'Invalid amount'
+    }
+
+    if (amount.valueBN.gt(nonStakedTokens)) {
+      return 'Insufficient balance'
+    }
+
+    return null
+  }, [amount])
+
+  // Calculate percentages
+  const nonStakedPct = round(pct(nonStakedTokens, stakeToken.balanceBN))
+  const stakedPct = 100 - nonStakedPct
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <Info
+        css={`
+          margin-top: ${3 * GU}px;
+        `}
+      >
+        This action will create a proposal which can be voted on by staking
+        {stakeToken.symbol}. The action will be executable if the accrued total
+        stake reaches above the threshold.
+      </Info>
+      <Field
+        label="amount"
+        css={`
+          margin-top: ${2 * GU}px;
+        `}
+      >
+        <TextInput
+          value={amount.value}
+          onChange={handleAmountChange}
+          onFocus={() => handleEditMode(true)}
+          onBlur={() => handleEditMode(false)}
+          wide
+          ref={inputRef}
+          adornment={
+            <ButtonBase
+              css={`
+                margin-right: ${1 * GU}px;
+                color: ${theme.accent};
+              `}
+              onClick={handleMaxSelected}
+            >
+              MAX
+            </ButtonBase>
+          }
+          adornmentPosition="end"
+        />
+      </Field>
+      <Button
+        label="Support this proposal"
+        wide
+        type="submit"
+        mode="strong"
+        disabled={amount.valueBN.eq(new BN(0)) || Boolean(errorMessage)}
+      />
+      {errorMessage && (
+        <Info
+          css={`
```

Comment:
i think that brett suggested us to move the css to the end of the component props, could be a good line to follow